src/app/invoice/invoice-detail/invoice-detail.component.ts

```
@@ -0,0 +1,159 @@
+    this.localCurrency = this.currencyService.detectUserCurrency();
+  }
+
+  ngOnInit() {
+    this.loadExchangeRate(); // Load exchange rates when component initializes
+  }
+  loadExchangeRate() {
+    this.currencyService.getExchangeRates(this.localCurrency).subscribe(
+      data => {
+        this.exchangeRates = data;
+
+        if (this.exchangeRates && this.exchangeRates.rates) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)